contrib/checkpoint/checkpoint-restore-cri-test.sh

```diff
@@ -0,0 +1,170 @@
+	echo "--> Deleting all pods: "
+	crictl -t 5s rmp -fa | sed 's/^/----> \t/'
+	echo -n "--> Pulling base image ${TEST_IMAGE}: "
+	crictl pull "${TEST_IMAGE}"
+	POD_JSON=$(mktemp)
+	# adapt the log directory
+	jq ".log_directory=\"${TESTDIR}\"" "$TESTDATA"/sandbox_config.json >"$POD_JSON"
+	echo -n "--> Start pod: "
+	pod_id=$(crictl runp "$POD_JSON")
+	echo "$pod_id"
+	echo -n "--> Create container: "
+	ctr_id=$(crictl create "$pod_id" "$TESTDATA"/container_sleep.json "$POD_JSON")
+	echo "$ctr_id"
+	echo -n "--> Start container: "
+	crictl start "$ctr_id"
+	lines_before=$(crictl logs "$ctr_id" | wc -l)
+	# changes file system to see if changes are included in the checkpoint
+	echo "--> Modifying container rootfs"
+	crictl exec "$ctr_id" touch /home/counter/testfile
+	crictl exec "$ctr_id" rm /home/counter/.bash_logout
+	echo -n "--> Checkpointing container: "
+	crictl -t 10s checkpoint --export="$TESTDIR"/cp.tar "$ctr_id"
+	echo "--> Cleanup container: "
+	crictl rm -f "$ctr_id" | sed 's/^/----> \t/'
+	echo "--> Cleanup pod: "
+	crictl rmp -f "$pod_id" | sed 's/^/----> \t/'
+	echo "--> Cleanup images: "
+	crictl rmi "${TEST_IMAGE}" | sed 's/^/----> \t/'
+	echo -n "--> Start pod: "
+	pod_id=$(crictl runp "$POD_JSON")
+	echo "$pod_id"
+	# Replace original container with checkpoint image
+	RESTORE_JSON=$(mktemp)
+	jq ".image.image=\"$TESTDIR/cp.tar\"" "$TESTDATA"/container_sleep.json >"$RESTORE_JSON"
+	echo -n "--> Create container from checkpoint: "
+	# This requires a larger timeout as we just deleted the image and
+	# pulling can take some time.
+	ctr_id=$(crictl -t 20s create "$pod_id" "$RESTORE_JSON" "$POD_JSON")
+	echo "$ctr_id"
+	rm -f "$RESTORE_JSON" "$POD_JSON"
+	echo -n "--> Start container from checkpoint: "
+	crictl start "$ctr_id"
+	sleep 1
+	lines_after=$(crictl logs "$ctr_id" | wc -l)
+	if [ "$lines_before" -ge "$lines_after" ]; then
+		echo "number of lines after checkpointing ($lines_after) " \
+			"should be larger than before checkpointing ($lines_before)"
+		false
+	fi
+	# Cleanup
+	echo "--> Cleanup images: "
+	crictl rmi "${TEST_IMAGE}" | sed 's/^/----> \t/'
+	echo -n "--> Verifying container rootfs: "
+	crictl exec "$ctr_id" ls -la /home/counter/testfile
+	if crictl exec "$ctr_id" ls -la /home/counter/.bash_logout >/dev/null 2>&1; then
+		echo "error: file /etc/exports should not exist but it does"
+		exit 1
+	fi
+	SUCCESS=1
+}
+
+function test_from_oci() {
+	echo "--> Deleting all pods: "
+	crictl -t 5s rmp -fa | sed 's/^/----> \t/'
+	echo -n "--> Pulling base image ${TEST_IMAGE}: "
+	crictl pull "${TEST_IMAGE}"
+	echo -n "--> Start pod: "
+	pod_id=$(crictl runp "$TESTDATA"/sandbox_config.json)
+	echo "$pod_id"
+	echo -n "--> Create container: "
+	ctr_id=$(crictl create "$pod_id" "$TESTDATA"/container_sleep.json "$TESTDATA"/sandbox_config.json)
+	echo "$ctr_id"
+	echo -n "--> Start container: "
+	crictl start "$ctr_id"
+	echo -n "--> Checkpointing container: "
+	crictl -t 10s checkpoint --export="$TESTDIR"/cp.tar "$ctr_id"
```

Comment:
Would it be possible to extend the `ctr container checkpoint` and `ctr containers restore` commands with `--export` and `--import` options? It would make it easier to use/test this functionality without `crictl`.

Reply:
Not as part of this PR at least. The `ctr` command and the CRI implementation are different code paths.